docs/docs/examples.md

```diff
@@ -0,0 +1,317 @@
+```
+
+Since both header and payload are fed into the signature, we need to prepare this signature input:
+
+```kotlin
+val plainSignatureInput = JwsSigned.prepareJwsSignatureInput(header, payload)
+```
+
+Now, everything is ready to be signed:
+
+```kotlin
+val signature = signer.sign(plainSignatureInput.encodeToByteArray()).signature //TODO: handle error
+JwsSigned(header, payload, signature, plainSignatureInput).serialize() // this we can verify on jwt.io 
+```
+
+As can be seen, a `JwsSigned` takes header, payload, signature, and the plain signature input as parameters.
+The reason for keeping this fourth parameter is convenience and efficiency: For one, you need this input to serialize a
+`JwsSigned`, so it would be a waste to discard it. After parsing a `JswSigned` from its serialized form, you also need the 
+`plainSignatureInput` to verify everything was signed correctly.
+
+
+## Creating a `CoseSigned` Object
+
+!!! info  inline end
+    This example requires the _Supreme_ KMP crypto provider and _Indispensable Cosef_.
+
+In this example, we'll again start with an ephemeral P-256 signer:
+
+```kotlin
+val signer = Signer.Ephemeral {
+    ec { curve = ECCurve.SECP_256_R_1 }
+}.getOrThrow() //TODO handle error properly
+```
+
+Next up, we'll create a header and payload:
+
+```kotlin
+//set KID + algorithm
+val protectedHeader = CoseHeader(
+    algorithm = signer.signatureAlgorithm.toCoseAlgorithm().getOrElse { TODO() },
+    kid = signer.publicKey.didEncoded.encodeToByteArray()
+)
+
+val payload = byteArrayOf(0xC, 0xA, 0xF, 0xE)
+```
+
+Both of these are signature inputs, so we'll construct a `CoseSignatureInput` to sign.
+
+```kotlin
+val signatureInput = CoseSignatureInput(
+    contextString = "Signature1",
+    protectedHeader = ByteStringWrapper(protectedHeader),
+    externalAad = byteArrayOf(),
+    payload = payload,
+).serialize()
+```
+
+
+Now, everything is ready to be signed:
+
+```kotlin
+val signature = signer.sign(signatureInput).signature //TODO handle error
+
+val coseSigned = CoseSigned(
+    ByteStringWrapper(protectedHeader),
+    unprotectedHeader = null,
+    payload,
+    signature
+).serialize() // sadly, there's no cwt.io, but you can use cbor.me to explore the signed data
+```
+
+## Create and Parse a Custom-Tagged ASN.1 Structure
+
+!!! info inline end
+    This example requires only the _Indispensable_ module.
+
+This example illustrates how to encapsulate a custom ASN.1 encoding scheme to make it reusable and composable.
+
+### Definitions
+
+Let's say you are using ASN.1 as your wire format for interoperability with different frameworks and languages.
+This particular example demonstrates log reporting, i.e. the status of an operation, maybe from a smartcard is sent off-device.
+
+!!! note inline end
+    Such constraints may seem artificial, but when bandwidth is low, a compact representation is key.
+
+A log message is an implicitly tagged ASN.1 structure with APPLICATION tag `1337` and sequence semantics.
```

Comment:
"compact representation is key" -> chooses a tag that doesn't fit into a single byte 😈